### ISSUE TITLE
`.pdf`/`.txt` assets in `mdsvex`?

### ISSUE THREAD

**Issue by janosh** [bluwy/svelte-preprocess-import-assets#16]
Doesn't seem to work with unusual assets like `.pdf`/`.txt` in `mdsvex`.

[StackBlitz repro](https://stackblitz.com/edit/sveltejs-kit-template-default-ynbrga?file=src/routes/+page.md) steps:

Start the dev server, click on the `'link'`. 

> Not found: /robots.txt

I expected the file to open (in the same or another tab).

Tried to tweak the `urlFilter` and `sources` config but didn't help.

```ts
import relative_imports from 'svelte-preprocess-import-assets'

export default {
preprocess: [
    mdsvex({
      extensions: [`.svx`, `.md`],
    }),
    preprocess(),
    relative_imports({
      sources: [
        { tag: `img`, attribute: `src` },
        { tag: `link`, attribute: `href` },
      ],
      urlFilter: (url) => {
        const out = /\.(png|jpg|gif|webp|pdf|svg)$/.test(url)
        if (url.endsWith(`.pdf`)) console.log(url, out)

        return out
      },
    }),
  ],
}
```

**Comment by bluwy** [comment]
I think you're looking for:

```js
relative_import({
  sources: (defaultSources) => {
    return [
      ...defaultSources,
      {
        tag: 'a',
        srcAttributes: ['href'],
        filter({ attributes }) {
          return /\.(txt|pdf)$/.test(attributes.href)
        }
      }
    ]
  }
})
```

The preprocessor doesn't transform `a` tags by default since that could interfere with routing. Also I noticed a bug with the `filter` API (https://github.com/bluwy/svelte-preprocess-import-assets/issues/17), but the above workaround should do for now.

**Comment by janosh** [comment]
@bluwy Thanks for the code snippet! It gives me a `TypeError` 

```
Cannot read properties of undefined (reading 'length')
    at node_modules/.pnpm/svelte-preprocess-import-assets@0.2.5_svelte@3.55.1/node_modules/svelte-preprocess-import-assets/dist/index.mjs:200:36
```

```js
let getAttributes = function() {
  if (!lazyAttributes) {
    lazyAttributes = {};
    node.attributes.forEach((attr) => {
    // error is raised here: attr.value.length
      if (attr.value.length > 1 && attr.value[0].type !== "Text")
        return;
      lazyAttributes[attr.name] = attr.value[0].raw;
    });
  }
  return lazyAttributes;
}
```

**Comment by bluwy** [comment]
That looks like a different AST handling issue with the preprocessor. With the fix it works fine in the repro though. Do you have an example mdsvex file where this happens?

**Comment by janosh** [comment]
I don't actually know which file causes this error. The stack trace only lists node_modules files. How would I find the offending content file?

**Comment by janosh** [comment]
Sorry for the long silence here. Just had another look at this. The code that's triggering this error is spreading `target` into `a` tags:

```svelte
<script lang="ts">
  const target = { target: `_blank`, rel: `noreferrer` }
</script>

<address>
  <a href="href" {...target}>
    Link 1
  </a>
  <a href="href" {...target}>
    Link 2
  </a>
  <a href="href" {...target}>
    Link 3
  </a>
</address>
```

Changing to this fixes the error:

```svelte
<address>
  <a href="href" target="_blank" rel="noreferrer">
    Link 1
  </a>
  <a href="href" target="_blank" rel="noreferrer">
    Link 2
  </a>
  <a href="href" target="_blank" rel="noreferrer">
    Link 3
  </a>
</address>
```

Injecting a `console.log` like you suggested for `attr` and `filename` was very helpful! The object that triggered the error:

```
attr {
  start: 529,
  end: 539,
  type: 'Spread',
  expression: Node {
    type: 'Identifier',
    start: 533,
    end: 538,
    loc: SourceLocation { start: [Position], end: [Position] },
    name: 'links'
  }
}
```

**Comment by bluwy** [comment]
Thanks! I didn't expect a `Spread` would be considered an attribute too. I'll fix that on up and cut a patch

**Comment by bluwy** [comment]
Fixed in v0.2.6
